src/main/java/teammates/common/util/LogHelper.java

```diff
@@ -0,0 +1,193 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import teammates.common.exception.InvalidParametersException;
+
+import com.google.appengine.api.log.AppLogLine;
+import com.google.appengine.api.log.LogQuery;
+import com.google.appengine.api.log.LogServiceFactory;
+import com.google.appengine.api.log.RequestLogs;
+import com.google.appengine.api.log.LogService.LogLevel;
+
+/**
+ * An utility to fetch logs from GAE server.
+ */
+public class LogHelper {
+    /**
+     * Maximum number of versions to query.
+     * The current value will include the current version and its 5 preceding versions.
+     */
+    private static final int MAX_VERSIONS_TO_QUERY = 6;
+    
+    /**
+     * A flag to decide whether to include application logs in result or not.
+     */
+    private static final boolean INCLUDE_APP_LOG = true;
+    
+    /**
+     * Affects the internal strategy to get logs. It doesn't affect the result.
+     */
+    private static final int BATCH_SIZE = 1000;
+    private static final LogLevel MIN_LOG_LEVEL = LogLevel.INFO;
+    public static final int SEARCH_TIME_INCREMENT = 2 * 60 * 60 * 1000;  // two hours in milliseconds
+    
+    private static LogQuery query;
+    private static Long endTime;
+    private static List<String> versionList;
```

Comment:
Non-final variables in a fully static class indicates a design smell. In other words, this is a singleton class, implemented as a fully static class (not recommended). 
I still think you need a `LogReader` class that can take a `AdminLogQuery` object, execute it, and return the result. Alternatively, you can move the `LogReader` stuff (i.e. methods that depend on GAE api) into `GaeAdminApi` class. What's left would be the `AdminLogQuery` class.